api-2.2/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl_2_2Test.java

```diff
@@ -90,14 +116,166 @@ public void shouldReturnNullWhenGetConditionByWrongUuid() {
+		assertThat(results.iterator().next().getPatient().getGivenName(), equalTo(PATIENT_GIVEN_NAME));
+		assertThat(results.iterator().next().getPatient().getFamilyName(), equalTo(PATIENT_FAMILY_NAME));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByOnsetDate() {
+		String testDate = "2017-01-12";
+		
+		DateParam onsetDate = new DateParam("eq" + testDate);
+		Collection<Condition> results = dao.searchForConditions(null, null, null, null, onsetDate, null, null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.size(), equalTo(3));
+		assertThat(results.iterator().next().getOnsetDate().toString(), startsWith(testDate));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByRecordedDate() {
+		String testDate = "2016-01-12";
+		
+		DateParam recordedDate = new DateParam("eq" + testDate);
+		Collection<Condition> results = dao.searchForConditions(null, null, null, null, null, null, recordedDate, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.size(), equalTo(1));
+		assertThat(results.iterator().next().getDateCreated().toString(), startsWith(testDate));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByClinicalStatusActive() {
+		TokenOrListParam listParam = new TokenOrListParam();
+		listParam.add(new TokenParam("active"));
+		Collection<Condition> results = dao.searchForConditions(null, null, null, listParam, null, null, null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.size(), equalTo(5));
+		assertThat(results.iterator().next().getClinicalStatus(), equalTo(ConditionClinicalStatus.ACTIVE));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByClinicalStatusInactive() {
+		TokenOrListParam listParam = new TokenOrListParam();
+		listParam.add(new TokenParam("inactive"));
+		Collection<Condition> results = dao.searchForConditions(null, null, null, listParam, null, null, null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.size(), equalTo(1));
+		assertThat(results.iterator().next().getClinicalStatus(), equalTo(ConditionClinicalStatus.INACTIVE));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByClinicalStatusAll() {
+		TokenOrListParam listParam = new TokenOrListParam();
+		listParam.add(new TokenParam("active"));
+		listParam.add(new TokenParam("inactive"));
+		Collection<Condition> results = dao.searchForConditions(null, null, null, listParam, null, null, null, null);
+		
+		assertThat(results, notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.size(), equalTo(6));
+	}
+	
+	@Test
+	public void searchForPatients_shouldReturnConditionByCode() {
+		TokenOrListParam listParam = new TokenOrListParam();
+		listParam.add(new TokenParam("CD41003")); // for concept_id=5497
```

Comment:
Where does this CD41003 value come from?

Reply:
Thanks! I haven't looks through the standard dataset in a while, so I forgot about that.